flang/lib/Lower/CallInterface.cpp

```diff
@@ -828,21 +851,22 @@ class Fortran::lower::CallInterfaceImpl {
 };
 
 template <typename T>
-void Fortran::lower::CallInterface<T>::buildImplicitInterface(
+void Fortran::lower::CallInterface<T>::determineInterface(
+    bool isImplicit,
     const Fortran::evaluate::characteristics::Procedure &procedure) {
   CallInterfaceImpl<T> impl(*this);
-  impl.buildImplicitInterface(procedure);
-}
-
-template <typename T>
-void Fortran::lower::CallInterface<T>::buildExplicitInterface(
-    const Fortran::evaluate::characteristics::Procedure &procedure) {
-  CallInterfaceImpl<T> impl(*this);
-  impl.buildExplicitInterface(procedure);
+  if (isImplicit)
+    impl.buildImplicitInterface(procedure);
+  else
+    impl.buildExplicitInterface(procedure);
+  if constexpr (std::is_same_v<T, Fortran::lower::CalleeInterface>) {
```

Comment:
Yes. This was just a quick, dirty way to avoid adding more "unreachable" methods to Caller and SignatureBuilder.

I can extend those interfaces of course (to make the C++ compiler happy) or I can just add a FIXME type of comment. I think you're suggesting the comment route here.